src/Illuminate/Support/UrlQueryParameters.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Illuminate\Support;
+
+use Illuminate\Contracts\Support\Arrayable;
+use Stringable;
+
+class UrlQueryParameters implements Arrayable, Stringable
+{
+    /**
+     * Constructor.
+     *
+     * @param  array  $parameters
+     * @return void
+     */
+    public function __construct(
+        protected array $parameters = []
+    ) {
+    }
+
+    /**
+     * Parse a query string.
+     *
+     * @param  string|null  $query
+     * @return static
+     */
+    public static function parse(?string $query): static
+    {
+        if (is_null($query)) {
+            return new static;
+        }
+
+        parse_str(
+            Str::of(urldecode($query))
```

Comment:
The `parse_url` method [internally performs `urldecode`](https://www.php.net/manual/en/function.parse-str.php#:~:text=All%20variables%20created%20(or%20values%20returned%20into%20array%20if%20second%20parameter%20is%20set)%20are%20already%20urldecode()d.) on the query string, so passing a decoded query string causes it to be decoded twice, leading to unexpected results:

```php
parse_str('test=1%2B2', $params);            // ["test" => "1+2"]

parse_str(urldecode('test=1%2B2'), $params); // ["test" => "1 2"]
```

Reply:
Ah excellent thanks for this @hafezdivandari! Just patched and updated a test case to ensure this is fixed. Appreciate the review.